contracts/facets/EmbalmerFacet.sol

```diff
@@ -34,267 +34,237 @@ contract EmbalmerFacet {
+                selectedArchaeologists[i].doubleHashedKeyShare
+            ] = selectedArchaeologists[i].archAddress;
+            s.archaeologistSarcophagi[selectedArchaeologists[i].archAddress].push(sarcoId);
         }
 
-        // Create the sarcophagus object and store it in AppStorage
-
-        // Sarco will continue to be in Active state past its resurrection window
-        // if no archaeologist ever unwraps it and it's never cleaned, buried or accused.
-        // Failure can be inferred by simply making sure to check if the Sarco has
-        // expired when its state is Active.
-        // Note this is also true for the Resurrecting state, if at leaset one of the
-        // archaeologists did unwrap before it was too late.
-        s.sarcophagi[sarcoId] = LibTypes.Sarcophagus({
-            name: sarcophagus.name,
-            state: LibTypes.SarcophagusState.Active,
-            minShards: sarcophagus.minShards,
-            resurrectionTime: sarcophagus.resurrectionTime,
-            maximumRewrapInterval: sarcophagus.maximumRewrapInterval,
-            arweaveTxIds: arweaveTxIds,
-            embalmer: msg.sender,
-            recipientAddress: sarcophagus.recipient,
-            archaeologists: cursedArchaeologists
-        });
-
-        // Add the identifier to the necessary data structures
+        // update sarcophagus-specific convenience lookup structures
         s.sarcophagusIdentifiers.push(sarcoId);
         s.embalmerSarcophagi[msg.sender].push(sarcoId);
-        s.recipientSarcophagi[sarcophagus.recipient].push(sarcoId);
+        s.recipientSarcophagi[sarcophagusParams.recipientAddress].push(sarcoId);
 
-        // Transfer the total fees amount + protocol fees in sarco token from the embalmer to this contract
+        // Transfer totalDiggingFees and the protocolFees in SARCO from embalmer to this contract
         uint256 protocolFees = LibUtils.calculateProtocolFees(totalDiggingFees);
-
-        // Add the create sarcophagus protocol fee to the total protocol fees in storage
         s.totalProtocolFees += protocolFees;
+        s.sarcoToken.transferFrom(msg.sender, address(this), totalDiggingFees + protocolFees);
 
-        s.sarcoToken.transferFrom(
-            msg.sender,
-            address(this),
-            totalDiggingFees + protocolFees
-        );
-
-        // Emit the event
         emit CreateSarcophagus(
             sarcoId,
-            sarcophagus.name,
-            sarcophagus.resurrectionTime,
+            sarcophagusParams.name,
+            sarcophagusParams.resurrectionTime,
             msg.sender,
-            sarcophagus.recipient,
-            cursedArchaeologists,
+            sarcophagusParams.recipientAddress,
+            sarcophagus.cursedArchaeologistAddresses,
             totalDiggingFees,
             protocolFees,
             arweaveTxIds
         );
 
-        // Return the index of the sarcophagus
+        // return index of sarcophagus in sarcophagusIdentifiers
```

Comment:
are we using `sarcophagusIdentifiers`? I didn't see it exposed in viewstatefacet